src/main/java/se/kth/sda/skeleton/posts/PostController.java

```diff
@@ -1,8 +1,57 @@
 package se.kth.sda.skeleton.posts;
 
-/*
-    @TODO create the methods needed to implement the API.
-    Don't forget to add necessary annotations.
- */
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import se.kth.sda.skeleton.posts.exeption.*;
+import se.kth.sda.skeleton.posts.PostService;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RequestMapping("/posts")
+@RestController
 public class PostController {
+    PostRepository postRepository;
+    PostService postService;
+
+    @Autowired
+    public PostController(PostRepository postRepository, PostService postService) {
+        this.postRepository = postRepository;
+        this.postService = postService;
+    }
+    /* create a new post */
+    @PostMapping("/")
+    public ResponseEntity<Post> createPost(@Valid @RequestBody Post post) {
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .body(postService.createPost(post));
+    }
+
+    /* return all posts*/
+    @GetMapping("/")
+    public ResponseEntity <List<Post>>listAllPost() {
+        return ResponseEntity.ok(postService.listAllPosts());
```

Comment:
I find the 2 methods `listAllPost()` and `listAllPosts()` confusing

Reply:
I think for clarity maybe they should rename service methods to for example:
`listAllPostsService`
What do you think?

Reply:
Also, now I saw, definitely rename this method to plural, so: `listAllPosts`

Reply:
Also, you should not add the slash there, just leave it empty:
` @GetMapping("")`
Because like this the req need to be sent to `/posts/` and we don't want that

Reply:
Now removed slash symbol from first two methods, now only ("")